src/Microsoft.HttpRepl/VerbosityLogger.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using Microsoft.Repl.ConsoleHandling;
+
+namespace Microsoft.HttpRepl
+{
+    internal class VerbosityLogger
```

Comment:
Having a concrete class named VerbosityLogger referenced throughout the code is feeling awkward to me.  To any consuming code, it should be just a Logger, or better yet, an ILogger.  The implementation can be a WhateverLogger, but that should be resolved at composition time (or maybe an ILoggerFactory for more dynamic acquisition cases).

Also, the acquisition of the logger, which is currently a static method, means that you have hidden dependencies strewn in the consuming code.  It's not following good DI principles.

Reply:
This basically stems from a really poor attempt by me at pulling this particular feature out of a larger effort to enable a sort of "Verbosity Context" for commands that would allow a command to turn on verbosity (with an option or otherwise) and have it flow to any child commands or components. VerbosityLogger was kind of a stripped down version of the work in progress there, but I did a poor job shoehorning it in. 

I've removed that, and simplified the implementation here (at least, I think). I still don't like the tight coupling of specifying the verbosity as it goes into the ApiConnection, but its not awful and its better than specifying the VerbosityLogger in those places.